### PR TITLE
Support conversion from Strict OOXML

### PR DESCRIPTION
**Pull request by selalerercapitolis**

### CODE REVIEW

**Comment by pjfanning** [comment]
thanks - I had not considered adding this support here but I'll consider it, might take me a few days to have a look

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.github.pjfanning</groupId>
   <artifactId>excel-streaming-reader</artifactId>
-  <version>2.3.6-SNAPSHOT</version>
+  <version>2.3.7-SNAPSHOT</version>
```
Comment (pjfanning) [inline pom.xml:6]: can you remove this? - I'll worry about the version numbers.

File: src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverter.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+import javax.xml.stream.events.Namespace;
+import javax.xml.stream.events.StartElement;
+import javax.xml.stream.events.XMLEvent;
+import org.apache.poi.hssf.usermodel.HSSFAnchor;
+import org.apache.poi.hssf.usermodel.HSSFDataBarFormatting;
+import org.apache.poi.hssf.usermodel.HSSFDateUtil;
+
+public class OoXmlStrictConverter implements AutoCloseable {
+
+    private static final XMLEventFactory XEF = XMLEventFactory.newInstance();
```
Comment (pjfanning) [inline src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverter.java:29]: there is a class called org.apache.poi.util.XMLHelper has methods to create the XML input and output factories - can you use those?

File: src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverter.java [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    private static final XMLInputFactory XIF = XMLInputFactory.newInstance();
+    private static final XMLOutputFactory XOF = XMLOutputFactory.newInstance();
+    private static final QName CONFORMANCE = new QName("conformance");
+    private static final Properties mappings;
+
+    static {
+        XOF.setProperty(XMLOutputFactory.IS_REPAIRING_NAMESPACES, true);
```
Comment (pjfanning) [inline src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverter.java:36]: this line won't be needed if you use the XMLHelper method to create an output factory

File: src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverterInputStream.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        END_STATE,
+        XML_ENTRY_START,
+        IN_NON_XML_ENTRY,
+        IN_XML_ENTRY,
+    }
+
+    private static class ByteArrayOutputStreamExposed extends ByteArrayOutputStream {
+
+        public byte[] getBytes() {
+            return buf;
```
Comment (pjfanning) [inline src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverterInputStream.java:179]: this is more efficient than https://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayOutputStream.html#toByteArray() ?

File: src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverterUtils.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                    props.setProperty(vals[0], vals[1]);
+                } else if(vals.length == 1) {
+                    props.setProperty(vals[0], "");
+                }
+
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        System.out.println("loaded mappings entries=" + props.size());
```
Comment (pjfanning) [inline src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverterUtils.java:66]: can you remove the println?

File: pom.xml (hunk 1 above)
Comment (selalerercapitolis) [reply pom.xml]: Done

File: src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverter.java (hunk 2 above)
Comment (selalerercapitolis) [reply src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverter.java]: Done

File: src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverter.java (hunk 3 above)
Comment (selalerercapitolis) [reply src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverter.java]: Done

File: src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverterUtils.java (hunk 5 above)
Comment (selalerercapitolis) [reply src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverterUtils.java]: Done

File: src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverterInputStream.java (hunk 4 above)
Comment (selalerercapitolis) [reply src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverterInputStream.java:179]: `toByteArray()` creates a copy. Probably not that important, performance-wise, considering we're doing IO and unzipping and zipping unnecessarily but just seemed weird to clone the array when all I need to do is read from it.

**Comment by pjfanning** [comment]
I published a new 2.3.6-SNAPSHOT release.

**Pull request merged**